Content.Server/Sticky/Systems/StickySystem.cs

```diff
@@ -20,15 +21,55 @@ public sealed class StickySystem : EntitySystem
     [Dependency] private readonly SharedHandsSystem _handsSystem = default!;
     [Dependency] private readonly SharedInteractionSystem _interactionSystem = default!;
     [Dependency] private readonly SharedAppearanceSystem _appearance = default!;
+    [Dependency] private readonly EntityLookupSystem _lookup = default!;
+    [Dependency] private readonly ILogManager _log = default!;
+    private ISawmill _sawmill = default!;
 
     private const string StickerSlotId = "stickers_container";
 
     public override void Initialize()
     {
         base.Initialize();
+        SubscribeLocalEvent<TryStickOnSpawnComponent, MapInitEvent>(OnAutoStickMapInit);
+        SubscribeLocalEvent<TransferStickySurfaceOnSpawnBehaviorComponent, OnSpawnFromSpawnEntitiesBehaviourEvent>(TransferStickySurfaceFromDestroyedEntity);
+
         SubscribeLocalEvent<StickyComponent, StickyDoAfterEvent>(OnStickFinished);
         SubscribeLocalEvent<StickyComponent, AfterInteractEvent>(OnAfterInteract);
         SubscribeLocalEvent<StickyComponent, GetVerbsEvent<Verb>>(AddUnstickVerb);
+
+        _sawmill = _log.GetSawmill("sticky");
+    }
+
+    private void OnAutoStickMapInit(EntityUid uid, TryStickOnSpawnComponent component, MapInitEvent args)
+    {
+        StickyComponent? sticky = null;
+        if (component.Shot || !Resolve(uid, ref sticky))
+            return;
+
+        var stuck = false;
+        foreach (var entity in _lookup.GetEntitiesInRange(Transform(uid).MapPosition, component.Range))
+        {
+            // check whitelist and blacklist
+            if (sticky.Whitelist != null && !sticky.Whitelist.IsValid(entity))
+                continue;
+            if (sticky.Blacklist != null && sticky.Blacklist.IsValid(entity))
+                continue;
+
+            StickToEntity(uid, entity, null, sticky);
+            component.Shot = true;
+            stuck = true;
+            break;
+        }
+
+        if (!stuck)
+        {
+            // entity was unable to stick on anything; draw it as unsticked
+            if (TryComp(uid, out AppearanceComponent? appearance))
+            {
+                _appearance.SetData(uid, StickyVisuals.IsStuck, false, appearance);
+            }
+            _sawmill.Warning($"Stickable entity '{EntityManager.ToPrettyString(uid)}' was supposed to stick on other entity nearby when spawn but couldn't find anything to stick on");
```

Comment:
I need to update appearance because it could be set as stuck which results in entity drawn as stuck when it's actually not. Warning could be useful to see which entities can't stick on start

Reply:
I dont really understand the issue? As far as i understand the appearance will only update if the item is sucessfully sticked? If the sticking fails but the sprite still updates that is an issue with StickToEntity that shouldnt be bandaided here.

Reply:
> I dont really understand the issue? As far as i understand the appearance will only update if the item is sucessfully sticked? If the sticking fails but the sprite still updates that is an issue with StickToEntity that shouldnt be bandaided here.

No, it updates only if it didn't stick. Without it, posters that were unable stick will be drawn as stuck. If I will make their appearance in prototypes drawn as rolled - then it will be impossible to understand how this poster looks.

Reply:
okay I can delete these lines now as I just used IconComponent to fix the issue